crates/codegen/utils/src/internal/formatting.rs

```diff
@@ -69,7 +69,7 @@ fn backup_raw_file(file_path: &PathBuf, unformatted: &str) -> Result<PathBuf> {
 
 fn generate_header(file_path: &PathBuf) -> Result<String> {
     let warning_line =
-        "This file is generated automatically by infrastructure scripts. Please don't edit by hand.";
+        "This file is @generated automatically by infrastructure scripts. Please don't edit by hand.";
```

Comment:
Like I said above, I don't feel strongly about this change, as it's very minor. This marker is slowly being adopted and had some adoption in the Rust space as well (i.e. Cargo/Rustfmt); if a future IDE extension adopts it to detect the marker and warns the user if they make a change or at some point GitHub will detect this rather than rely on explicit `linguist-generated` tags then we'd be already setup.

We don't have to drive the adoption and since this doesn't bring any tangible benefit at the moment I can back this one out, if you prefer :+1:

Reply:
In that case, I suggest removing it, as we do want to format them.
I agree on the benefit of tags like `@generated`. They are useful for many cases, like files that won't be read by humans, or specific code blocks that can be broken if they are formatted (rust has `#[rustfmt::skip]` I think for these cases). But for the files here, we do want to keep them formatted, linted, and treated as "real" source files, since they are user facing, and they would be read and debugged by users, so the same rules should apply here. Please let me know if you see otherwise.

Reply:
The default Rustfmt's behaviour is to format these, however there's no tangible benefit for the tag right now.

Maybe we could detect the tag and warn in-editor when we modify them? I found myself modifying the generated code by mistake few times after Ctrl+F'ing there only to have them being ovewritten a few times now, however we can leave this discussion for later and outside of this PR :+1: I reverted the change.